sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -116,9 +157,49 @@ object UnsupportedOperationChecker {
           throwError("Commands like CreateTable*, AlterTable*, Show* are not supported with " +
             "streaming DataFrames/Datasets")
 
-        case m: MapGroupsWithState if collectStreamingAggregates(m).nonEmpty =>
-          throwError("(map/flatMap)GroupsWithState is not supported after aggregation on a " +
-            "streaming DataFrame/Dataset")
+        // mapGroupsWithState
+        case m: FlatMapGroupsWithState if m.isStreaming && m.isMapGroupsWithState =>
```

Comment:
No such syntax. I can use something like `case m@MapGroupsWithState(_, _, ..., _, outputMode, _) => if outputMode == Update`. But doesn't save the codes.